system_test/aest_heavy_sync_SUITE.erl

```diff
@@ -0,0 +1,184 @@
+
+end_per_suite(_Config) -> ok.
+
+
+%=== TEST CASES ================================================================
+
+%% This is keypair of an account set in genesis config file
+%% (see https://github.com/aeternity/epoch/blob/master/data/aecore/.genesis/accounts.json),
+%% so beneficiary configuration in epoch.yaml (mining > beneficiary param) does not matter.
+patron() ->
+    #{ pubkey => <<206,167,173,228,112,201,249,157,157,78,64,8,128,168,111,29,73,187,68,75,98,241,26,158,187,100,187,207,235,115,254,243>>,
+       privkey => <<230,169,29,99,60,119,207,87,113,50,157,51,84,179,188,239,27,197,224,50,196,61,112,182,211,90,249,35,206,30,183,77,206,167,173,228,112,201,249,157,157,78,64,8,128,168,111,29,73,187,68,75,98,241,26,158,187,100,187,207,235,115,254,243>>
+     }.
+
+many_spend_txs(Cfg) ->
+    setup_nodes([#{ name    => node1,
+                    peers   => [node2, node3],
+                    backend => aest_docker,
+                    source  => {pull, "aeternity/epoch:local"},
+                    config  => #{persist => false}
+                  },
+                 #{ name    => node2,
+                    peers   => [node1],
+                    backend => aest_docker,
+                    source  => {pull, "aeternity/epoch:local"},
+                    config  => #{persist => true}
+                  },
+                 #{ name    => node3,
+                    peers   => [node1],
+                    backend => aest_docker,
+                    source  => {pull, "aeternity/epoch:local"},
+                    config  => #{persist => true},
+                    mining  => #{autostart => false}
+                  }], Cfg),
+    
+    %% Load many transactions to node3 and then sync transactions to a mining node.
+    %% Transactions sync is faster that transactions posting and limit on http clients no issue
+    
+    %% This means we always create micro blocks in first generation (after genesis block)
+    %% That is a sligtly unrealistic situation, but if we first mine a few key blocks,
+    %% then this test takes ages.
+
+    TxsPerMB = aec_governance:block_gas_limit() div (aec_governance:tx_base_gas(spend_tx) + 1720),
+    ct:log("We can put ~p Txs in a micro block\n", [TxsPerMB]),
+
+    start_node(node3, Cfg),
+    wait_for_startup([node3], 0, Cfg),
+
+    Patron = patron(),
+
+    %% We assume 300 spend per Micro block. 2700 txs takes 9 micro blocks and should be under
+    %% 9*3 seconds (half a minute). Expected to have at most 10 micro blocks, otherwise
+    %% we are not efficient enough
+    TxHashes = add_spend_txs(node3, Patron, 2700, 1),
+
+    start_node(node1, Cfg),
+    wait_for_startup([node1], 0, Cfg),
+    LastTxHash = lists:last(TxHashes),
+
+    %% Check that the valid transactions made it to the chain.
+    #{node1 := [{_Hash, Height}]} = 
+        wait_for_value({txs_on_chain, [LastTxHash]},
+                   [node1], 10 * ?MINING_TIMEOUT, [{key_blocks, 0} | Cfg]),  
```

Comment:
I do not see where `key_blocks` option is used.

Reply:
It is to _not_ wait for the next keyblock, thus we will have a large "current generation" which was what triggers the timeout.

Reply:
It is in aest_nodes:wait_for_value({txs_on_chain,... but probably that only made it to master recently (PR1670)